pretyping/tacred.ml

```diff
@@ -321,11 +321,11 @@ let compute_consteval allowed_reds env sigma ref u =
           let nbfix = Array.length lv in
           (if nbfix = 1 then
             let names = [|Some (ref,u)|] in
-            try check_fix_reversibility env sigma ref u labs l names fix
+            try check_fix_reversibility env sigma ref u n labs l names fix
             with Elimconst -> NotAnElimination
           else
-            let labs, l, names = invert_names allowed_reds env sigma lastref lastu names i in
-            try check_fix_reversibility env sigma lastref lastu labs l names fix
+            let labs', l', names = invert_names allowed_reds env sigma lastref lastu names i in
```

Comment:
Why '?

Reply:
Because there are other `labs` and `l` and it seems clearer to name differently the labs/l of the original name (e.g. boo in your example) and the labs/l of the immediate name of the fixpoint (e.g. even in your example), especially due to the fact that the `n` passed on the next line is the length of the initial `labs`.